doc/developers/maintainer.rst

```diff
@@ -141,36 +149,50 @@ Making a release
    The wheel building workflow is managed by GitHub Actions and the results be browsed at:
    https://github.com/scikit-learn/scikit-learn/actions?query=workflow%3A%22Wheel+builder%22
 
-   Once all works, you can proceed with tagging. Create the tag and push it (if
+.. note::
+
+  Before building the wheels, make sure that the ``pyproject.toml`` file is
+  up to date and using the oldest version of ``numpy`` for each Python version
+  to avoid `ABI <https://en.wikipedia.org/wiki/Application_binary_interface>`_
+  incompatibility issues. Moreover, a new line have to be included in the
+  ``pyproject.toml`` file for each new supported version of Python.
```

Comment:
I think that this information is not necessary if we decided to move on with `oldest-supported-numpy` #18900.